zipkin-api.yaml

```diff
@@ -0,0 +1,259 @@
+
+        - name: limit
+          in: query
+          default: 10
+          type: integer
+          description: |
+            Maximum number of traces to return. Defaults to 10
+      responses:
+        200:
+          description: OK
+          schema:
+            $ref: "#/definitions/ListOfTracesLists"
+  /trace/{traceId}:
+    parameters:
+      - name: traceId
+        in: path
+        required: true
+        description: the 64-bit hex-encoded id of the trace as a path parameter.
+        type: string
+    get:
+      responses:
+        200:
+          description: OK
+          schema:
+            description: An array of spans (annotations?)
+            type: array
+            items:
+              $ref: "#/definitions/Annotation"
+        404:
+           description: "`traceId` not found"
+  /spans:
+    post:
+      description: Uploads a list of spans in json format
+      consumes:
+        - application/json
+        - application/x-thrift
+      parameters:
+        - name: span
+          in: body
+          description: A new span
+          required: true
+          schema:
+            $ref: "#/definitions/TraceList"
+      responses:
+        202:
+          description: Created
+  /dependencies:
+    get:
+      description: |
+        Returns dependency links derived from spans.
+
+        Span names are in lowercase, rpc method for example. Conventionally,
+        when the span name isn't known, name = "unknown".
+      parameters:
+        - name: endTs
+          in: query
+          description: |
+            only return links from spans where Span.timestamp  are at or before
+            this time in epoch milliseconds.
+          required: true
+          type: integer
+        - name: loopback
+          in: query
+          description: |
+            only return links where all Span.timestamp are at or after (endTs -
+            * lookback) in milliseconds. Defaults to endTs, limited to a system
+            parameter QUERY_LOOKBACK
+          type: string
+      responses:
+        200:
+          description: OK
+          schema:
+            type: array
+            title: ListOfDependencyLinks
+            items:
+              $ref: "#/definitions/DependencyLink"
+definitions:
+  Annotation:
+    type: object
+    properties:
+      endpoint:
+        type: object
+        properties:
+          serviceName:
+            type: string
+          ipv4:
+            type: string
+          port:
+            type: integer
+      timestamp:
+        type: integer
+      value:
+        type: string
+  BinaryAnnotation:
+    type: object
+    properties:
+      key:
+        type: string
+      value:
+        type: string
+      endpoint:
+        $ref: '#/definitions/Endpoint'
+  Endpoint:
+    title: Endpoint
+    type: object
+    properties:
+      serviceName:
+        type: string
+      ipv4:
+        type: string
+      port:
+        type: integer
+  TraceList:
```

Comment:
ListOfSpans (for upload) or Trace (for query). Difference:

ListOfSpans - possibly different trace ids, in no particular order
Trace - List of spans who have the same trace id... Spans in the trace, and annotations in a span are sorted ascending by timestamp. ie first event should be first in the spans list.